### ISSUE TITLE
Multivariate hypergeometric function

### ISSUE THREAD

**Issue by Cyril13** [markrogoyski/math-php#372]
Hi, is possible to create this ?
Like I have 30 balls : 5 red, 10 blue and 15 green.
On a sample of 10 balls, what's the probability to have 1 of each ? At least 1 of each ? At least 1 red and 2 blue ? 
Etc ....

Thanks a lot !

**Comment by Beakerboy** [comment]
Do you mean the [Hypergeometric_function](https://en.wikipedia.org/wiki/Hypergeometric_function) or the [Hypergeometric_distribution](https://en.wikipedia.org/wiki/Hypergeometric_distribution)?

**Comment by Cyril13** [comment]
I don't really know, an algo to solve the balls problem.
Thanks !

**Comment by Beakerboy** [comment]
I think you're looking for the later [wikipedia](https://en.wikipedia.org/wiki/Hypergeometric_distribution#Multivariate_hypergeometric_distribution)

We do not, but we do have the combinations and permutations that can be used to solve these problems [Combinatorics](https://github.com/markrogoyski/math-php#probability---combinatorics)

**Comment by Beakerboy** [comment]
Do you have example data that we could use to compare to?

**Comment by Beakerboy** [comment]
#373

**Comment by Cyril13** [comment]
Hi Kevin, I don't really have data, I was thinking of an algo to resolve the Balls problems. Not the one with "at least X balls of green color", but the one with for exemple "exactly 1 red and 2 green", then for the rest ("at least") it just will be enumeration of a finite number of possibilities.

Thanks a lot !

________________________________
De : Kevin Nowaczyk <notifications@github.com>
Envoyé : mercredi 11 mars 2020 00:14
À : markrogoyski/math-php <math-php@noreply.github.com>
Cc : Cyril13 <cyr-13@hotmail.fr>; Author <author@noreply.github.com>
Objet : Re: [markrogoyski/math-php] Multivariate hypergeometric function (#372)


Do you have example data that we could use to compare to?

**Comment by Beakerboy** [comment]
Look at patch #373 and let me know if that would suit your needs. Feel free to comment on that patch discussion.

**Comment by Cyril13** [comment]
Thanks a lot !

**Comment by markrogoyski** [comment]
@Cyril13,

This is how the code works.
```php
// Suppose there are 5 black, 10 white, and 15 red marbles in an urn.
$quantities = [5, 10, 15];

/// If six marbles are chosen without replacement, the probability that exactly two of each color are chosen is:
$choices = [2, 2, 2]

$distribution = new Hypergeometric($quantities);
$probability = $distribution->pmf($choices);  // 0.0795756
```

It is equivalent to doing the following in R:
```R
dmvhyper(c(2,2,2), c(5,10,15), 6)
[1] 0.0795756
```

**Comment by Cyril13** [comment]
Thanks again !

________________________________
De : Mark Rogoyski <notifications@github.com>
Envoyé : vendredi 13 mars 2020 07:13
À : markrogoyski/math-php <math-php@noreply.github.com>
Cc : Cyril13 <cyr-13@hotmail.fr>; Mention <mention@noreply.github.com>
Objet : Re: [markrogoyski/math-php] Multivariate hypergeometric function (#372)


@Cyril13<https://github.com/Cyril13>,

This is how the code works.

// Suppose there are 5 black, 10 white, and 15 red marbles in an urn.
$quantities = [5, 10, 15];

/// If six marbles are chosen without replacement, the probability that exactly two of each color are chosen is:
$choices = [2, 2, 2]

$distribution = new Hypergeometric($quantities);
$probability = $distribution->pmf($choices);  // 0.0795756

It is equivalent to doing the following in R:

dmvhyper(c(2,2,2), c(5,10,15), 6)
[1] 0.0795756

**Comment by markrogoyski** [comment]
This has been merged and will go out with the next release. Thanks for suggesting the feature.

**Comment by markrogoyski** [comment]
This has been released in [v0.61.0](https://github.com/markrogoyski/math-php/releases/tag/v0.61.0).

Thanks again for the feature request suggestion!
